test/unit/functions/noSelfAnnotation.spec.ts

```diff
@@ -62,3 +62,25 @@ test("explicit this parameter respected over @noSelf", () => {
         export const result = foo(1);
     `.expectToMatchJsResult();
 });
+
+test("respect noSelfInFile over noImplicitSelf", () => {
```

Comment:
Still missing a test where the declaration of the function is in a different file than the call to that function,